src/pint/models/noise_model.py

```diff
@@ -556,6 +561,125 @@ def pl_dm_cov_matrix(self, toas):
         return np.dot(Fmat * phi[None, :], Fmat.T)
 
 
+class PLSWNoise(NoiseComponent):
+    """Model of solar wind DM variations as radio frequency-dependent noise with a
+    power-law spectrum.
+
+    Commonly used as perturbations on top of a deterministic solar wind model.
+
+
+    Parameters supported:
+
+    .. paramtable::
+        :class: pint.models.noise_model.PLSWNoise
+
+    Note
+    ----
+    Ref: Lentati et al. 2014, MNRAS 437(3), 3004-3023.
+    See also: Hazboun et al. 2019 and Susurla et al. 2024.
+
+    """
+
+    register = True
+    category = "pl_SW_noise"
+
+    introduces_correlated_errors = True
+    is_time_correlated = True
+
+    def __init__(
+        self,
+    ):
+        super().__init__()
+
+        self.add_param(
+            floatParameter(
+                name="TNSWAMP",
+                units="",
+                aliases=[],
+                description="Amplitude of power-law SW DM noise in tempo2 format",
+                convert_tcb2tdb=False,
+            )
+        )
+        self.add_param(
+            floatParameter(
+                name="TNSWGAM",
+                units="",
+                aliases=[],
+                description="Spectral index of power-law "
+                "SW DM noise in tempo2 format",
+                convert_tcb2tdb=False,
+            )
+        )
+        self.add_param(
+            floatParameter(
+                name="TNSWC",
+                units="",
+                aliases=[],
+                description="Number of SW DM noise frequencies.",
+                convert_tcb2tdb=False,
+            )
+        )
```

Comment:
I presume this only supports the inverse-squared radial variation of the solar wind. This corresponds to `SWP=2` and `SWM=1`. 

If this is correct, please add a validation step somewhere (I suggest in `get_noise_basis` or `get_noise_weights`) to ensure that this condition is met. If not, this component will be inconsistent with `SolarWindDispersion`.

Reply:
hi @abhisrkckl ,
i am trying to follow up with all of my open PRs since my term just ended. 😄 
i don’t think this necessarily requires inverse-square solar wind variations, but i agree that we need some way to ensure the same SWP used in the noise run is being plugged into the timing model. i’m not entirely sure how to do this, but i will keep looking into it.

Reply:
One possibility is to check the SWM and SWP parameters and use them to determine how this mode will behave. Or if all of the SWM and SWP values are supported you can throw an error if the condition is not met. The main point is to avoid inconsistency between models.